Tensile/KernelWriter.py

```diff
@@ -529,37 +566,56 @@ def makeSubIterSchedule(self, kernel, localReadCode, iteration, pointerCode, wai
     if isinstance(waitCode, Code.WaitCnt):
       # Set the waitCount, based on the new iter schedule
       lgkmcnt = 0 # most conservative
+      localReads = 0
+      localWrites = 0
       if kernel["EnableMatrixInstruction"]:
-        localReads  = localReadCode.countType(Code.LocalReadInst)
-        localWrites = 0
+        lrvwA = 1
+        lrvwB = 1
+        if not kernel["TransposeLDS"] or (kernel["TransposeLDS"] and (kernel["ProblemType"]["TLUA"] or kernel["ProblemType"]["TLUB"])):
+          if kernel["ProblemType"]["DataType"].isBFloat16():
+            lrvwA = lrvwA / 2 if kernel["ProblemType"]["TLUA"] else 1
+            lrvwB = lrvwB / 2 if kernel["ProblemType"]["TLUB"] else 1
+          if kernel["ProblemType"]["DataType"].isHalf():
+            lrvwA = lrvwA / 4 if kernel["ProblemType"]["TLUA"] else 1
+            lrvwB = lrvwB / 4 if kernel["ProblemType"]["TLUB"] else 1
+        numA = kernel["InnerUnroll"]*(kernel["MIWaveTile"][0] // lrvwA) \
+            // self.localReadInstructionA.numOffsets
+        numB = kernel["InnerUnroll"]*(kernel["MIWaveTile"][1] // lrvwB) \
+            // self.localReadInstructionB.numOffsets
+        readsPerIter = numA + numB
```

Comment:
Should this be same as above? If so, these values should probably be saved as kernel params.

numA = kernel["InnerUnroll"]*(kernel["MIWaveTile"][0] * numReadPerVectorA) // tensorParametersA["localReadInstruction"].numOffsets
      numB = kernel["InnerUnroll"]*(kernel["MIWaveTile"][1] * numReadPerVectorB) // tensorParametersB["localReadInstruction"].numOffsets
      readsPerIter = numA + numB if self.numReadsIterCoalesced == 1 else 0